azure/services/virtualmachineimages/images.go

```diff
@@ -47,8 +47,59 @@ func New(auth azure.Authorizer) (*Service, error) {
 	}, nil
 }
 
+// GetDefaultLinuxImage returns the default image spec for a Linux node.
+func (s *Service) GetDefaultLinuxImage(ctx context.Context, location, k8sVersion string) (*infrav1.Image, error) {
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "virtualmachineimages.Service.GetDefaultLinuxImage")
+	defer done()
+
+	// First try Azure Linux, then Ubuntu.
+	defaultImage, err := s.GetDefaultAzureLinuxImage(ctx, location, k8sVersion)
```

Comment:
This change can add a couple of API calls to this common code path if it has to fall back to Ubuntu. But they all ultimately call `getSKUAndVersion` which implements a cache, so in practice it shouldn't cause many new round-trips.